src/ts/reader.ts

```diff
@@ -48,3 +50,23 @@ export class AllowListReader {
     return readStorage(this.allowList, this.reader, "numSolvers", []);
   }
 }
+
+/**
+ * A class for attaching the storage reader contract to the GPv2Settlement contract
+ * for providing additional storage reading methods.
+ */
+export class SettlementReader {
+  constructor(
+    public readonly settlement: Contract,
+    public readonly reader: Contract,
+  ) {}
+
+  /**
+   * Read and return filled amounts for a list of orders
+   */
+  public filledAmountsForOrders(orderUids: string[]): Promise<number> {
```

Comment:
I'd see this function like this:
```suggestion
  public filledAmountsForOrders(orderUids: BytesLike[]): Promise<BigNumber[]> {
```
The input type changes as discussed in the other comment, the output type is actually a vector of big numbers and not a number.

Note that you need to `import BigNumber` and `BytesLike` for this to work.